### PR TITLE
World sim builder add model with transformation

### PR DESCRIPTION
**Pull request by hongkai-dai**
This is used when I simulate the cup flipping example, that the mounting transform for our Schunk gripper is different from the iiwa_ee_frame defined in the URDF.

---
This change is [Reviewable](https://reviewable.io/reviews/robotlocomotion/drake/7928)

### CODE REVIEW

**Comment by hongkai-dai** [comment]
+@sammy-tri for feature review please.

---

Review status: 0 of 3 files reviewed at latest revision, all discussions resolved.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/robotlocomotion/drake/7928#-:-L4IN7JXwd7PnlJN2nzg:bjquaum)*

**Comment by sammy-tri** [comment]
:lgtm: with a couple minor issues.  +@sherm1 for platform review

---

Reviewed 3 of 3 files at r1.
Review status: all files reviewed at latest revision, 3 unresolved discussions.

---

*[manipulation/util/world_sim_tree_builder.h, line 72 at r1](https://reviewable.io:443/reviews/robotlocomotion/drake/7928#-L4MTbkGiAvVx73rWfvQ:-L4MTbkHsY-fReZNrklL:bwgwtx2) ([raw file](https://github.com/robotlocomotion/drake/blob/afb32e356ae6c94eed0394f092820dd2ab6c31fc/manipulation/util/world_sim_tree_builder.h#L72)):*
> ```objc
> 
>   /// Adds a model instance specified by its model name, @p model_name, to the
>   /// `RigidBodyTree` being built. The model instance is connected to the
> ```

"to the a new frame" sounds a little awkward (remove "the"?)

---

*[manipulation/util/world_sim_tree_builder.h, line 80 at r1](https://reviewable.io:443/reviews/robotlocomotion/drake/7928#-L4MTnpzlWtdDZAgO08P:-L4MTnpzlWtdDZAgO08Q:bjwq80w) ([raw file](https://github.com/robotlocomotion/drake/blob/afb32e356ae6c94eed0394f092820dd2ab6c31fc/manipulation/util/world_sim_tree_builder.h#L80)):*
> ```objc
>   /// loaded via a call to StoreModel().
>   /// The function will search for the body with name @p weld_to_body_name, on
>   /// the model with @p model_instance_id.
> ```

BTW it seems a little odd to separate these parameters since they are closely relates.  maybe make `model_instance_id` adjacent to `weld_to_body_name`?

---

*[manipulation/util/test/sim_diagram_builder_test.cc, line 248 at r1](https://reviewable.io:443/reviews/robotlocomotion/drake/7928#-L4MUMuquAJMn9uorIq_:-L4MUMurvJ5qSdCorNdu:b4br6kd) ([raw file](https://github.com/robotlocomotion/drake/blob/afb32e356ae6c94eed0394f092820dd2ab6c31fc/manipulation/util/test/sim_diagram_builder_test.cc#L248)):*
> ```cpp
>         "frame_schunk_to_iiwa_link_ee", iiwa.at(i).instance_id);
>     EXPECT_TRUE(CompareMatrices(schunk_frame->get_transform_to_body().matrix(),
>               transform_schunk_to_iiwa_link_ee.matrix()));
> ```

odd indent?

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/robotlocomotion/drake/7928#-:-L4MUSIcDxnO4ZRCx1SD:b-stbfo)*

**Comment by hongkai-dai** [comment]
Review status: 0 of 3 files reviewed at latest revision, 3 unresolved discussions.

---

*[manipulation/util/world_sim_tree_builder.h, line 72 at r1](https://reviewable.io:443/reviews/robotlocomotion/drake/7928#-L4MTbkGiAvVx73rWfvQ:-L4MkBLyQHLXlyxZDNeY:boyexb3) ([raw file](https://github.com/robotlocomotion/drake/blob/afb32e356ae6c94eed0394f092820dd2ab6c31fc/manipulation/util/world_sim_tree_builder.h#L72)):*
<details><summary><i>Previously, sammy-tri (Sam Creasey) wrote…</i></summary><blockquote>

"to the a new frame" sounds a little awkward (remove "the"?)
</blockquote></details>

Done. Thanks!

---

*[manipulation/util/world_sim_tree_builder.h, line 80 at r1](https://reviewable.io:443/reviews/robotlocomotion/drake/7928#-L4MTnpzlWtdDZAgO08P:-L4Mkq5NSufPNal1vRX9:b-896fix) ([raw file](https://github.com/robotlocomotion/drake/blob/afb32e356ae6c94eed0394f092820dd2ab6c31fc/manipulation/util/world_sim_tree_builder.h#L80)):*
<details><summary><i>Previously, sammy-tri (Sam Creasey) wrote…</i></summary><blockquote>

BTW it seems a little odd to separate these parameters since they are closely relates.  maybe make `model_instance_id` adjacent to `weld_to_body_name`?
</blockquote></details>

Done.

---

*[manipulation/util/test/sim_diagram_builder_test.cc, line 248 at r1](https://reviewable.io:443/reviews/robotlocomotion/drake/7928#-L4MUMuquAJMn9uorIq_:-L4MlKPd6QgMbLi9b-lB:b-896fix) ([raw file](https://github.com/robotlocomotion/drake/blob/afb32e356ae6c94eed0394f092820dd2ab6c31fc/manipulation/util/test/sim_diagram_builder_test.cc#L248)):*
<details><summary><i>Previously, sammy-tri (Sam Creasey) wrote…</i></summary><blockquote>

odd indent?
</blockquote></details>

Done.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/robotlocomotion/drake/7928)*

**Comment by sherm1** [comment]
Platform :lgtm: except for some naming/commenting clarifications needed (can discuss f2f if you want).

---

Reviewed 3 of 3 files at r2.
Review status: all files reviewed at latest revision, 3 unresolved discussions.

---

*[manipulation/util/world_sim_tree_builder.h, line 75 at r2](https://reviewable.io:443/reviews/robotlocomotion/drake/7928#-L4Mq0Ro5gtW2aWCWqn-:-L4Mq0Ro5gtW2aWCWqn0:btqwjbg) ([raw file](https://github.com/robotlocomotion/drake/blob/4c5b602a3577230fa0ed584e263555d3c18faf3b/manipulation/util/world_sim_tree_builder.h#L75)):*
> ```objc
>   /// a new frame constructed within this function. This new frame will be
>   /// welded to a body of name @p weld_to_body_name, with a transformation to
>   /// this body as @p transform frame_to_body. The new frame is named @p
> ```

BTW this should use an unambiguous notation X_FB or X_BF, or if you want to use the longer name you should give it an unambiguous definition. There is apparently an assumption that there is a unique "body frame" on every body (which makes sense). Then I think we are gluing another frame F to that body. Then is there also some distinguished but unspecified "model frame" M that gets welded to F to enforce M=F?

---

*[manipulation/util/test/sim_diagram_builder_test.cc, line 228 at r2](https://reviewable.io:443/reviews/robotlocomotion/drake/7928#-L4MmebrL5qst-cXB4da:-L4MmebrL5qst-cXB4db:briwin7) ([raw file](https://github.com/robotlocomotion/drake/blob/4c5b602a3577230fa0ed584e263555d3c18faf3b/manipulation/util/test/sim_diagram_builder_test.cc#L228)):*
> ```cpp
>   }
> 
>   const Eigen::Isometry3d transform_schunk_to_iiwa_link_ee =
> ```

BTW consider naming the frames and then making this X_AB for suitable A and B (optional since this is a test case but would set a good example).

---

*[manipulation/util/test/sim_diagram_builder_test.cc, line 239 at r2](https://reviewable.io:443/reviews/robotlocomotion/drake/7928#-L4Mny0Mi1I-x3cqOhzI:-L4Mny0N7pGIGNXTWBHn:b-bw07y5) ([raw file](https://github.com/robotlocomotion/drake/blob/4c5b602a3577230fa0ed584e263555d3c18faf3b/manipulation/util/test/sim_diagram_builder_test.cc#L239)):*
> ```cpp
>     int id = tree_builder->AddModelInstanceToFrame(
>         "wsg", "iiwa_link_ee", iiwa.at(i).instance_id,
>         "frame_schunk_to_iiwa_link_ee", transform_schunk_to_iiwa_link_ee,
> ```

BTW I find this name hard to parse. I guess it is a frame rigidly fixed to body "iiwa_link_ee" that is the intended mounting fixture for the gripper? My preference would be to define frames in comments, like "frame S is fixed to the end effector and provides the location and orientation at which a schunk gripper should be mounted". Then the frame name could be "iiwa_link_ee_S" or something like that. Then you don't have to try to squeeze the whole definition into the name. (For your consideration.)

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/robotlocomotion/drake/7928#-:-L4MrXgt19wdEIXX36DD:bj35uyt)*

**Comment by hongkai-dai** [comment]
Review status: 0 of 3 files reviewed at latest revision, 3 unresolved discussions.

---

*[manipulation/util/world_sim_tree_builder.h, line 75 at r2](https://reviewable.io:443/reviews/robotlocomotion/drake/7928#-L4Mq0Ro5gtW2aWCWqn-:-L4Mxrcyd1BlErrokrdP:b-896fix) ([raw file](https://github.com/robotlocomotion/drake/blob/4c5b602a3577230fa0ed584e263555d3c18faf3b/manipulation/util/world_sim_tree_builder.h#L75)):*
<details><summary><i>Previously, sherm1 (Michael Sherman) wrote…</i></summary><blockquote>

BTW this should use an unambiguous notation X_FB or X_BF, or if you want to use the longer name you should give it an unambiguous definition. There is apparently an assumption that there is a unique "body frame" on every body (which makes sense). Then I think we are gluing another frame F to that body. Then is there also some distinguished but unspecified "model frame" M that gets welded to F to enforce M=F?
</blockquote></details>

Done.

---

*[manipulation/util/test/sim_diagram_builder_test.cc, line 228 at r2](https://reviewable.io:443/reviews/robotlocomotion/drake/7928#-L4MmebrL5qst-cXB4da:-L4My_5rfXn8Or_l9qVY:b-896fix) ([raw file](https://github.com/robotlocomotion/drake/blob/4c5b602a3577230fa0ed584e263555d3c18faf3b/manipulation/util/test/sim_diagram_builder_test.cc#L228)):*
<details><summary><i>Previously, sherm1 (Michael Sherman) wrote…</i></summary><blockquote>

BTW consider naming the frames and then making this X_AB for suitable A and B (optional since this is a test case but would set a good example).
</blockquote></details>

Done.

---

*[manipulation/util/test/sim_diagram_builder_test.cc, line 239 at r2](https://reviewable.io:443/reviews/robotlocomotion/drake/7928#-L4Mny0Mi1I-x3cqOhzI:-L4MyzxdoOS_0M0Ra6VZ:b-896fix) ([raw file](https://github.com/robotlocomotion/drake/blob/4c5b602a3577230fa0ed584e263555d3c18faf3b/manipulation/util/test/sim_diagram_builder_test.cc#L239)):*
<details><summary><i>Previously, sherm1 (Michael Sherman) wrote…</i></summary><blockquote>

BTW I find this name hard to parse. I guess it is a frame rigidly fixed to body "iiwa_link_ee" that is the intended mounting fixture for the gripper? My preference would be to define frames in comments, like "frame S is fixed to the end effector and provides the location and orientation at which a schunk gripper should be mounted". Then the frame name could be "iiwa_link_ee_S" or something like that. Then you don't have to try to squeeze the whole definition into the name. (For your consideration.)
</blockquote></details>

Done.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/robotlocomotion/drake/7928)*

**Comment by hongkai-dai** [comment]
-(status: curate commits before merging) 

---

Review status: 0 of 3 files reviewed at latest revision, all discussions resolved.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/robotlocomotion/drake/7928#-:-L4NfHOJXFdLp7R-i5rZ:b-l70wzf)*

**Comment by sherm1** [comment]
Thanks! Looks great.

---







*Comments from [Reviewable](https://reviewable.io:443/reviews/robotlocomotion/drake/7928#-:-L4NhQpBUJSguOUZ8EUI:b-es8gva)*
